v2/tools/generator/internal/codegen/pipeline/add_operator_spec.go

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/config"
+)
+
+const AddOperatorSpecStageID = "addOperatorSpec"
+
+func AddOperatorSpec(configuration *config.Configuration, idFactory astmodel.IdentifierFactory) Stage {
+	return MakeStage(
+		AddOperatorSpecStageID,
+		"Adds the property 'OperatorSpec' to all Spec types that require it",
+		func(ctx context.Context, state *State) (*State, error) {
+			defs := state.Types()
+			result := make(astmodel.Types)
+
+			propInjector := astmodel.NewPropertyInjector()
+			resources := astmodel.FindResourceTypes(defs)
+
+			for _, resource := range resources {
+				resolved, err := defs.ResolveResourceSpecAndStatus(resource)
+				if err != nil {
+					return nil, errors.Wrapf(err, "resolving resource spec and status for %s", resource.Name())
+				}
+
+				secrets, err := configuration.ObjectModelConfiguration.AzureGeneratedSecrets(resource.Name())
+				if err != nil {
+					// In this case, error is OK and just means we dont need to make an OperatorSpec type
+					continue
+				}
```

Comment:
Must use `IsNotConfiguredError()` in case other error kinds are returned; if `err != nil` and it's not a `NotConfiguredError`, abort the stage.